### ISSUE TITLE
password less repository sync

### ISSUE THREAD

**Issue by digtail** [kopia/kopia#3381]
I want to use kopia to make backups from some friends windows machines and store them on my sftp server / nas. I want the backups to be store on multiple redundant locations and have found an S3 BLOB storage provider I like. Administratively it would be easiest to set up the kopia ui on the windows machines and have them back up to my nas. My nas can then copy the repository to the S3 provider. There are tools to do this, rclone comes to mind. however, because kopia has a different repository layout for s3 and sftp, it is not possible to restore backups by connecting kopia to the s3 provider without first downloading the repository from s3. kopia has a way of syncing repositories between different providers (`kopia repository sync-to`). this however requires the repository password. I do not want to have my friends kopia repository passwords laying around on my nas, which would be required if I use `kopia repository sync-to s3` to upload the nas's repos to the s3 provider.

I do not want to set up kopia on the windows machines to make backups to multiple destinations. Afaik this is not possible to set up in the kopia ui. I really am not familiar with windows and I do not want to come up with hacky ways to work around this. I also want the backup process to be easy on the machines, and running a kopia sync-to would take up extra cpu and bandwith.

Is there a way to do this? can `kopia repository sync-to` be made to work without passwords? is there a simple way to convert between the s3 and sftp formats? Is there a much simpler solution I have not considered?

**Comment by basldfalksjdf** [comment]
I dont think this will ever be possible. Kopia is built with encryption as a basic requirement from the ground up. There is no other way Kopia could `sync-to` to another repository the way you want without being able to read the repository contents, which would require password.

`I do not want to set up kopia on the windows machines to make backups to multiple destinations. Afaik this is not possible to set up in the kopia ui. `

You can easily use KopiaUI to backup to multiple repositories. You can connect to as many repositories as you want. Just takes a few clicks.

**Comment by digtail** [comment]
okay, then I will have a closer look at the kopia ui and try again.

out of curiosity though, is it really impossible to transform a repository between storage formats without reencrypting it? I get that the way sync-to is implement requires reencryption, but is this requirement an implementation detail or a limit of how the different repository formats are structured?

thank you for the support and the project, it is greatly appreciated! :heart:
